server/core/src/atomic_apply_database_operations.cpp

```diff
@@ -0,0 +1,1274 @@
+            log::database::error("{} :: Database operation [{}] not supported on table [{}].", __func__, _db_op, _table_name);
+            return false;
+        }
+
+        return true;
+    } // is_db_operation_supported
+
+    auto throw_if_invalid_table_column(const std::string_view _column,
+                                       const std::string_view _table_name) -> void
+    {
+        const auto pred = [target = db_column_type{_table_name, _column}](const db_column_type& _v) noexcept
+        {
+            return _v == target;
+        };
+
+        if (std::none_of(std::begin(db_columns), std::end(db_columns), pred)) {
+            const auto msg = fmt::format("Invalid table column [{}.{}].", _table_name, _column);
+            log::database::error("{} :: {}", __func__, msg);
+            THROW(SYS_INVALID_INPUT_PARAM, msg);
+        }
+    } // throw_if_invalid_table_column
+
+    auto throw_if_invalid_conditional_operator(const std::string_view _operator) -> void
+    {
+        const auto pred = [&_operator](const std::string_view _v) noexcept
+        {
+            return _v == _operator;
+        };
+
+        if (std::none_of(std::begin(supported_operators), std::end(supported_operators), pred)) {
+            const auto msg = fmt::format("Invalid conditional operator [{}].", _operator);
+            log::database::error("{} :: {}", __func__, msg);
+            THROW(SYS_INVALID_INPUT_PARAM, msg);
+        }
+    } // throw_if_invalid_conditional_operator
+
+    auto current_timestamp() -> std::string
+    {
+        using std::chrono::system_clock;
+        using std::chrono::duration_cast;
+        using std::chrono::seconds;
+
+        return fmt::format("{:011}", duration_cast<seconds>(system_clock::now().time_since_epoch()).count());
+    } // current_timestamp
+
+    auto to_upper(std::string& _s) -> std::string&
+    {
+        std::transform(std::begin(_s), std::end(_s), std::begin(_s), [](unsigned char _c) {
+            return std::toupper(_c);
+        });
+
+        return _s;
+    } // to_upper
+
+    auto to_upper_copy(const std::string& _s) -> std::string
+    {
+        auto t = _s;
+        return to_upper(t);
+    } // to_upper_copy
+
+    template <typename Container>
+    auto at(const Container& _data, const std::string_view _column_name) -> std::string
+    {
+        const auto end = std::end(_data);
+        const auto iter = std::find_if(std::begin(_data), end, [&_column_name](const dml::column& _v) {
+            return _v.name == _column_name;
+        });
+
+        if (iter == end) {
+            THROW(SYS_INVALID_INPUT_PARAM, fmt::format("No data set for column [{}].", _column_name));
+        }
+
+        return iter->value;
+    } // at
+
+    template <typename Container>
+    auto value(const Container& _data,
+               const std::string_view _column_name,
+               const std::string_view _fallback_value) -> std::string
```

Comment:
changed the return type of this function to std::string_view because it allows this optimization. `_data` allows references to be returned and `_fallback_value` is a hard-coded value in the current implementation.